src/openforms/variables/models.py

```diff
@@ -100,3 +103,36 @@ def clean(self):
 
         if errors:
             raise ValidationError(errors)
+
+    def request_arguments(self, context: DataMapping) -> dict:
+        """Return a dictionary with keyword arguments for a
+        zgw_consumers.Service client request call.
+        """
+        headers = {
+            # no leading spaces and the value into the legal field-value codespace
+            header: value.format(**context).strip().encode("utf-8").decode("latin1")
+            for header, value in (self.headers or {}).items()
+        }
+        # before we go further
+        # assert headers are still valid after we put submmitter data in there
+        HeaderValidator()(headers)
+
+        escaped_values = {k: quote(str(v), safe="") for k, v in context.items()}
```

Comment:
additional thought - shouldn't this use `quote_plus` to ensure that spaces in input are also encoded with a `+` char? this also by default sets `safe=""` rather than the `quote` function which has a default `safe="/"`

Reply:
Cleaned it up even more thorough. No cutting corners anymore